smithy-aws-protocol-tests/model/awsJson1_0/required.smithy

```diff
@@ -0,0 +1,204 @@
+$version: "2.0"
+
+namespace aws.protocoltests.json10
+
+use aws.protocols#awsJson1_0
+use smithy.test#httpResponseTests
+
+apply OperationWithRequiredMembers @httpResponseTests(
+    [
+        {
+            id: "AwsJson10ClientErrorCorrectsWhenServerFailsToSerializeRequiredValues"
+            documentation: "Client error corrects when server fails to serialize required values."
+            appliesTo: "client"
+            tags: ["defaults", "error-correction"]
+            protocol: awsJson1_0
+            code: 200
+            bodyMediaType: "application/json"
+            headers: {"Content-Type": "application/x-amz-json-1.0"}
+            body: "{}"
+            params: {
+                requiredMembers: {
+                    requiredString: ""
+                    requiredBoolean: false
+                    requiredList: []
+                    requiredTimestamp: 0
+                    requiredBlob: ""
+                    requiredByte: 0
+                    requiredShort: 0
+                    requiredInteger: 0
+                    requiredLong: 0
+                    requiredFloat: 0
+                    requiredDouble: 0
+                    requiredMap: {}
+                    requiredBigInteger: 0
+                    requiredBigDecimal: 0
+                }
+            }
+        }
+    ]
+)
+
+apply OperationWithRequiredMembersWithDefaults @httpResponseTests(
+    [
+        {
+            id: "AwsJson10ClientErrorCorrectsWithDefaultValuesWhenServerFailsToSerializeRequiredValues"
+            documentation: "Client error corrects with default values when server fails to serialize required values."
+            appliesTo: "client"
+            tags: ["defaults", "error-correction"]
+            protocol: awsJson1_0
+            code: 200
+            bodyMediaType: "application/json"
+            headers: {"Content-Type": "application/x-amz-json-1.0"}
+            body: "{}"
+            params: {
+                requiredMembers: {
+                    requiredString: "hi"
+                    requiredBoolean: true
+                    requiredList: []
+                    requiredTimestamp: 1
+                    requiredBlob: "{}"
+                    requiredByte: 1
+                    requiredShort: 1
+                    requiredInteger: 10
+                    requiredLong: 100
+                    requiredFloat: 1
+                    requiredDouble: 1
+                    requiredMap: {}
+                    requiredBigInteger: 1
+                    requiredBigDecimal: 1
+                    requiredEnum: "FOO"
+                    requiredIntEnum: 1
+                }
+            }
+        }
+    ]
+)
+
+operation OperationWithRequiredMembers {
+    output := {
```

Comment:
let's flatten these into the output directly rather than nested. Clients won't typically error correct a missing struct since they'll generate required structs as nullable anyways.